File: src/java/sckgroup/conexion/Conexion.java
package sckgroup.conexion;

import java.sql.Connection;
import java.sql.DriverManager;

/**
 * Clase que se encarga de la conexion a la base de datos MySQL
 * del proyecto SCK-GROUP (modulo web)
 * @author Cindy Johana Ramirez
 * @version 1.0
 */
public class Conexion {

    static String url = "jdbc:mysql://localhost:3306/sck_group_db";
    static String usuario = "root";
    static String clave = "";

    /**
     * Metodo que conecta con la base de datos
     */
    public static Connection conectar() {
        Connection con = null;
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            con = DriverManager.getConnection(url, usuario, clave);
        } catch (Exception e) {
            System.out.println("Error de conexion: " + e.getMessage());
        }
        return con;
    }
}

File: config/conexion.php
<?php
// Archivo de conexion a la base de datos MySQL
// Proyecto: SCK-GROUP - Sistema de Creditos Kronos Group
// Autor: Cindy Johana Ramirez
// Ficha: 3070186

$host = "localhost";
$usuario = "root";
$clave = "";
$base_datos = "sck_group_db";

// Se conecta a la base de datos
$conexion = mysqli_connect($host, $usuario, $clave, $base_datos);

// Si la conexion falla se detiene el script
if (!$conexion) {
    http_response_code(500);
    echo json_encode(array("mensaje" => "Error al conectar con la base de datos."));
    die();
}

// Se configura el charset para caracteres especiales
mysqli_set_charset($conexion, "utf8");
?>
